xdsl/ir/core.py

```diff
@@ -1615,12 +1615,22 @@ def split_before(
     def get_operation_index(self, op: Operation) -> int:
         """Get the operation position in a block."""
         if op.parent is not self:
-            raise Exception("Operation is not a children of the block.")
+            raise Exception("Operation is not a child of the block.")
         for idx, block_op in enumerate(self.ops):
             if block_op is op:
                 return idx
         assert False, "Unexpected xdsl error"
 
+    def get_operation_at_index(self, idx: int) -> Operation:
+        """Get the operation by its position in its parent block."""
```

Comment:
Can you add that the complexity is in O(n)? Just as a warning to people to not use this too much.